service/src/main/java/com/codeforcommunity/processor/authenticated/ProtectedSchoolProcessorImpl.java

```diff
@@ -447,6 +447,50 @@ public ReportWithLibrary createReportWithLibrary(
         newReport.getParentSupport());
   }
 
+  @Override
+  public void adminUpdateReportWithLibrary(
+      JWTData userData, int schoolId, int reportId, UpsertReportWithLibrary req) {
+
+    if (!userData.isAdmin()) {
+      throw new AdminOnlyRouteException();
+    }
+    SchoolsRecord school = this.queryForSchool(schoolId);
+    if (school == null) {
+      throw new SchoolDoesNotExistException(schoolId);
+    }
+
+    // Save a record to the school_reports_with_libraries table
+    SchoolReportsWithLibrariesRecord newReport = db.selectFrom(SCHOOL_REPORTS_WITH_LIBRARIES)
+        .where(SCHOOL_REPORTS_WITH_LIBRARIES.ID.eq(reportId)).fetchOne();
+
+    if (newReport == null) {
+      throw new NoReportFoundException(schoolId);
+    }
+
+    newReport.setUserId(userData.getUserId());
+    newReport.setSchoolId(schoolId);
+    newReport.setNumberOfChildren(req.getNumberOfChildren());
+    newReport.setNumberOfBooks(req.getNumberOfBooks());
+    newReport.setMostRecentShipmentYear(req.getMostRecentShipmentYear());
+    newReport.setIsSharedSpace(req.getIsSharedSpace());
+    newReport.setHasInvitingSpace(req.getHasInvitingSpace());
+    newReport.setAssignedPersonRole(req.getAssignedPersonRole());
+    newReport.setAssignedPersonTitle(req.getAssignedPersonTitle());
+    newReport.setApprenticeshipProgram(req.getApprenticeshipProgram());
+    newReport.setTrainsAndMentorsApprentices(req.getTrainsAndMentorsApprentices());
+    newReport.setHasCheckInTimetables(req.getHasCheckInTimetables());
+    newReport.setHasBookCheckoutSystem(req.getHasBookCheckoutSystem());
+    newReport.setNumberOfStudentLibrarians(req.getNumberOfStudentLibrarians());
+    newReport.setReasonNoStudentLibrarians(req.getReasonNoStudentLibrarians());
+    newReport.setHasSufficientTraining(req.getHasSufficientTraining());
+    newReport.setTeacherSupport(req.getTeacherSupport());
+    newReport.setParentSupport(req.getParentSupport());
+
+    newReport.store();
+    newReport.refresh();
```

Comment:
❓ what does .refresh() do?

Reply:
We actually don't need refresh here, since we're not using the updated record (but it just fetches the fully updated record once it's stored). It was useful in a couple of other places for getting autogenerated columns.